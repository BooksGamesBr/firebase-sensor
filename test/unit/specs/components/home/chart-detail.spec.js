import { shallow } from '@vue/test-utils'
import ChartDetail from '@/components/home/chart-detail'
import DevLog from '../../../mocks/devLog.mock'

const propsData = {
  rooms: {
    room1: {
      color: '#FFCCBC',
      icon: 'glyphicon-bed',
      label: 'Room One',
      readings: {
        other: 0,
        temp: 1
      }
    }
  },
  category: {
    id: 'temp',
    forceMax: 15,
    forceMin: 0,
    fractionSize: 1,
    label: 'Temperature',
    order: 1,
    unit: 'Â°C'
  },
  numberOfHours: 24,
  readings: {
    room1: [
      { id: 'L00', time: 1514764800, temp: 12.1 },
      { id: 'L01', time: 1514764860, temp: 12.4 }
    ]
  }
}

describe('ChartDetail.vue', () => {
  it('shows a graph', () => {
    const wrapper = shallow(ChartDetail, {
      propsData,
      mocks: { DevLog },
      attachToDocument: true
    })
    expect(wrapper.html().replace(/nv-edge-clip-[\d]*/g, '')
      .replace(/nv-chart-[\d]*/g, '')
      .replace(/id="[^"]*"/g, '')
      .replace(/clip-path="[^"]*"/g, '')
      .replace(/d="[^"]*"/g, '')
      .replace(/class="[^"]*"/g, '')
      .replace(/transform="[^"]*"/g, '')).toMatchSnapshot()
  })
})
