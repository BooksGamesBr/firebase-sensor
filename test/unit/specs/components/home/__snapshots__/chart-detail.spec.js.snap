// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChartDetail.vue shows a graph 1`] = `
<div>
    <svg style="height: 500px;">
        <g>
            <g>
                <g>
                    <g>
                        <g>
                            <g>
                                <circle style="stroke-width: 2; fill: #FFCCBC; fill-opacity: 1; stroke: #FFCCBC;" r="5"></circle>
                                <text text-anchor="start" dy=".32em" dx="8"></text>
                            </g>
                        </g>
                    </g>
                </g>
                <g>
                    <g>
                        <rect></rect>
                    </g>
                    <g></g>
                    <g></g>
                    <g>
                        <g>
                            <defs>
                                <clipPath>
                                    <rect width="870" height="420"></rect>
                                </clipPath>
                            </defs>
                            <g>
                                <g>
                                    <g style="stroke-opacity: 1; stroke-width: 1.5; fill-opacity: 0.5; fill: #FFCCBC; stroke: #FFCCBC;">
                                        <path></path>
                                    </g>
                                </g>
                                <g>
                                    <g>
                                        <defs>
                                            <clipPath>
                                                <rect width="890" height="440"></rect>
                                            </clipPath>
                                        </defs>
                                        <g>
                                            <g>
                                                <g style="stroke-opacity: 1; fill-opacity: 0.5; fill: #FFCCBC; stroke: #FFCCBC;">
                                                    <path></path>
                                                    <path></path>
                                                </g>
                                            </g>
                                            <g></g>
                                        </g>
                                        <g></g>
                                    </g>
                                </g>
                            </g>
                        </g>
                    </g>
                    <g>
                        <g>
                            <g></g>
                        </g>
                    </g>
                </g>
                <g></g>
            </g>
        </g>
    </svg>
</div>
`;
